surfacers/bigquery/bigquery.go

```
@@ -0,0 +1,183 @@
+		colName := col.GetName()
+		val, err := convertToBqType(col.GetType(), em.Label(colName))
+		if err != nil {
+			return nil, fmt.Errorf("error occurred while parsing for field %v: %v", colName, err)
+		}
+		bqRowMap[colName] = val
+	}
+	return bqRowMap, nil
+}
+
+func (s *Surfacer) batchInsertRowsToBQ(ctx context.Context, inserter iInserter) {
+	chanLen := len(s.writeChan)
+	bigqueryTimeout := time.Duration(s.c.GetBigqueryTimeout()) * time.Second
+	bqctx := context.Background()
+	bqctx, cancel := context.WithTimeout(bqctx, bigqueryTimeout)
+	defer cancel()
+
+	for i := 0; i < chanLen; i += batchSize {
+		var results []*bqrow
+
+		for j := i; j < min(i+batchSize, chanLen); j++ {
+			em := <-s.writeChan
+
+			bqRowMap, err := s.parseBQCols(em)
+
+			if err == nil {
+				result := &bqrow{value: bqRowMap}
+				results = append(results, result)
+			} else {
+				s.l.Errorf("%v", err)
+			}
+		}
+
+		if len(results) > 0 {
+			if err := inserter.Put(bqctx, results); err != nil {
+				s.l.Errorf("failed uploading probe results to Bigquery due to surfacer: %v", err)
+			}
+		}
+	}
+}
+
+func (s *Surfacer) writeToBQ(ctx context.Context, inserter iInserter) {
+	bigqueryInsertionTime := s.c.GetBatchInsertionTime()
+	ticker := time.NewTicker(time.Duration(bigqueryInsertionTime) * time.Second)
+	for {
+		select {
+		case <-ctx.Done():
+			s.l.Infof("Context canceled, stopping the surfacer write loop")
+			s.batchInsertRowsToBQ(ctx, inserter)
+			return
+		case <-ticker.C:
+			s.batchInsertRowsToBQ(ctx, inserter)
+		}
+	}
+}
+
+func (s *Surfacer) init(ctx context.Context) error {
+	s.writeChan = make(chan *metrics.EventMetrics, s.c.GetMetricsBufferSize())
+
+	client, err := bigquery.NewClient(ctx, s.c.GetProjectName())
+	if err != nil {
+		s.l.Errorf("bigquery client can't be created: %v", err)
+		return err
+	}
+	defer client.Close()
+
+	inserter := client.Dataset(s.c.GetBigqueryDataset()).Table(s.c.GetBigqueryTable()).Inserter()
+	if inserter == nil {
```

Should you return an error if inserter is nil?